### DISCUSSION TITLE
Exception log for non premium users (ElmahCore)

### DISCUSSION THREAD

**Discussion by Surenthiran** [serenity-is/Serenity#6822] [Tricks and Samples]
Created new Wiki for Exception logging 

[https://github.com/serenity-is/Serenity/wiki/Exception-log-for-non-premium-users-(ElmahCore)](https://github.com/serenity-is/Serenity/wiki/Exception-log-for-non-premium-users-(ElmahCore))

**Comment by LSNicholls** [comment]
Hi -- thank you!  But the link didn't work for me... ? 
Pinging @Surenthiran - I really love Elmah and want to see how this would work...

**Reply by Surenthiran** [reply]
try this url 
[https://github.com/serenity-is/Serenity/wiki/Exception-log-for-non-premium-users-(ElmahCore)](https://github.com/serenity-is/Serenity/wiki/Exception-log-for-non-premium-users-(ElmahCore))

**Reply by LSNicholls** [reply]
Thank you so much!

**Comment by LSNicholls** [comment]
hi again @Surenthiran ,

 I had to change to 
`AddElmah <ElmahCore.Sql.SqlErrorLog>`
to get it to run.  (or a Using statement).

Other than that, as far as I know, I have followed the instructions.

But there are no rows showing in the Exceptions page (and the underlying table) no matter what type of error I force.  

Is there something else I need to do?

I used the most recent Serene template (6.9.6) and installed Elmah.Core, ElmahCore.Common, and ElmahCore.Sql, all are 2.1.2.

Thanks again in advance for your help.

**Reply by Surenthiran** [reply]
Sorry for outdated Wiki, After .netcore 8 update we also faced the same issue we need to add Emah into logger service. i have updated the Wiki please check.

**Comment by LSNicholls** [comment]
Thank you @Surenthiran.
> After .netcore 8 update

That's interesting, because my project was targetting v6.

Also, I did need to install ElmahCore.Sql from nuget along with Elmah.Core, so I have added that into the wiki.

And then it works, and looks great.  

Thank you so much!  I might even switch over another application that I wrote a custom logger for, in StartSharp, to use this instead!

I got a couple of messages about possible improvements, so I'm guessing we are on different versions of something somewhere, but NBD.

I may add some info about adjusting the CSS if I get there (I used to do this with Elmah, but I'm not sure how to do it in this version of Elmah) and also maybe either removing "About" on the menu or implementing it.

![image](https://github.com/serenity-is/Serenity/assets/88156108/2327e116-510b-401e-b307-87ff090bc94a)

**Reply by Surenthiran** [reply]
ElmahCore.Sql is needed if you are using SQL as storage. problem is all errors will not be added into elmah if we do not add those LoggerProvider. 

If you want seperate UI we can fetch data from Errors table and build on our own.

**Reply by LSNicholls** [reply]
> If you want seperate [sic] UI we can fetch data from Errors table and build on our own.

Of course, but this is not what I meant.  I actually like the default UI, just wanted to change it a little bit.  I did manage to do this with an iframe and some css to hide the top nav bar.  I am not satisfied with the implementation, so I'm not going to add into the wiki for this.

If I *did* add a module that worked directly from the table and built a based-on-Serenity UI, I would contribute that information to the wiki for sure, because I think it would be a less-trivial task to do it really well.

Thank you again for sharing!

![image](https://github.com/serenity-is/Serenity/assets/88156108/8fc7bf74-eb36-4737-9b37-ce1bb5749fe7)
